lib/order.rb

```diff
@@ -0,0 +1,28 @@
+require_relative 'takeaway'
+
+class Order
+
+  attr_reader :selected_dishes
+
+  def initialize 
+    @selected_dishes = []
+    @total = []
+  end 
+
+  def add_dish(dish)
+    @selected_dishes << dish
+  end 
+
+  def total
+    @selected_dishes.each { |item| @total << Takeaway.new.menu[item] } 
```

Comment:
much neater than what I went for here - works much better saving the order as an array of the names and then when you need the total calling the menu using those names